python/ray/dag/dag_node_operation.py

```diff
@@ -368,3 +505,72 @@ def _generate_actor_to_execution_schedule(
     for _, candidates in actor_to_candidates.items():
         assert len(candidates) == 0
     return actor_to_execution_schedule
+
+
+def _optimize_execution_schedule(
+    actor_to_execution_schedule: Dict[
+        "ray.actor.ActorHandle", List[_DAGOperationGraphNode]
+    ],
+    overlap_gpu_communication: bool,
+) -> Dict["ray.actor.ActorHandle", List[_DAGOperationGraphNode]]:
+    """
+    Optimize the execution schedule by overlapping computation and communication.
```

Comment:
Please add a comment explaining what the current algorithm is at a high level.

Reply:
The inline comments explains the algorithm. I added here as well.